src/Model/DICOMStructure.py

```diff
@@ -141,41 +133,37 @@ def __init__(self, study_id):
         """
         self.study_id = study_id
         self.study_description = None
-        self.series = []
+        self.series = {}
 
     def add_series(self, series):
         """
         Adds a Series object to the patient's list of series.
         :param series: A Series object.
         """
-        self.series.append(series)
+        self.series[series.series_id] = series
```

Comment:
This is not quite as problematic as study id vs study instance UID, because there isn't a Series ID element (series number is the equivalent). But series_uid would still be better